File: ASSIGNMENT9/python/Q.2.py
def has_zero_sum_subarray(arr):
    seen_sums = set()
    current_sum = 0
    for num in arr:
        current_sum += num
        if current_sum == 0 or current_sum in seen_sums:
            return True
        seen_sums.add(current_sum)
    return False

arr = [4, 2, -3, 1, 6]
print("Subarray with 0 sum exists:", has_zero_sum_subarray(arr))

File: ASSIGNMENT9/c/Q.2.c
#include <stdio.h>

int checkSubarrayWithZeroSum(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        int sum = 0;
        for (int j = i; j < n; j++) {
            sum += arr[j];
            if (sum == 0) {
                return 1;
            }
        }
    }
    return 0;
}

int main() {
    int arr[] = {4, 2, -3, 1, 6};
    int n = sizeof(arr) / sizeof(arr[0]);
    if (checkSubarrayWithZeroSum(arr, n)) {
        printf("Subarray with 0 sum exists\n");
    } else {
        printf("No subarray with 0 sum\n");
    }
    return 0;
}
